### PR TITLE
[SPARK-35116][SQL][TESTS] The generated data fits the precision of DayTimeIntervalType in spark

### PR DESCRIPTION
**Pull request by beliefer**
### What changes were proposed in this pull request?
The precision of `java.time.Duration` is nanosecond, but when it is used as `DayTimeIntervalType` in Spark, it is microsecond.
At present, the `DayTimeIntervalType` data generated in the implementation of `RandomDataGenerator` is accurate to nanosecond, which will cause the `DayTimeIntervalType` to be converted to long, and then back to `DayTimeIntervalType` to lose the accuracy, which will cause the test to fail. For example: https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/137390/testReport/org.apache.spark.sql.hive.execution/HashAggregationQueryWithControlledFallbackSuite/udaf_with_all_data_types/

### Why are the changes needed?
Improve `RandomDataGenerator` so that the generated data fits the precision of DayTimeIntervalType in spark.

### Does this PR introduce _any_ user-facing change?
'No'. Just change the test class.

### How was this patch tested?
Jenkins test.

### CODE REVIEW

File: sql/catalyst/src/test/scala/org/apache/spark/sql/RandomDataGenerator.scala [hunk 1]
```diff
@@ -275,8 +275,13 @@ object RandomDataGenerator {
...
         val seconds = rand.nextLong() % maxSeconds
+        // The precision of java.time.Duration is nanosecond, but when it is used as
+        // DayTimeIntervalType in Spark, it is microsecond. Here by following the behavior
+        // of DurationConverter to achieve consistency
         val nanoAdjustment = rand.nextLong() % 999999000
-        Duration.ofSeconds(seconds, nanoAdjustment)
+        val duration = Duration.ofSeconds(seconds, nanoAdjustment)
+        val micros = IntervalUtils.durationToMicros(duration)
+        IntervalUtils.microsToDuration(micros)
       })
```
Comment (MaxGekk) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/RandomDataGenerator.scala:285]: I think we can simplify the code:
```scala
      case DayTimeIntervalType => Some(() => Duration.of(rand.nextLong(), ChronoUnit.MICROS))
```

File: sql/catalyst/src/test/scala/org/apache/spark/sql/RandomDataGenerator.scala [hunk 2]
```diff
@@ -275,8 +275,13 @@ object RandomDataGenerator {
...
       case YearMonthIntervalType => Some(() => {
```
Comment (MaxGekk) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/RandomDataGenerator.scala:286]: The same here:
```scala
      case YearMonthIntervalType => Some(() => Period.ofMonths(rand.nextInt()))
```

**Review by MaxGekk — CHANGES_REQUESTED** [review]
The changes look good but I would simplify the code.

File: sql/catalyst/src/test/scala/org/apache/spark/sql/RandomDataGenerator.scala (hunk 1 above)
Comment (beliefer) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/RandomDataGenerator.scala:285]: OK

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
+1, LGTM. GA passed. Merging to master.
Thank you, @beliefer .

**Comment by beliefer** [comment]
@MaxGekk Thanks for your review.

**Pull request closed without merging**